### PR TITLE
[Refactor] Remove OCMock dependency in ZMSyncStrategy tests

### PR DESCRIPTION
**Pull request by typfel**
## What's new in this PR?

The `ZMSyncStrategy` tests needs to be improved.
- We only want to test the sync strategy logic but that's hard to do in isolation since the ZMSyncStrategy creates all request strategies.
- `ZMSyncStrategy` tests rely on `OCMock` which doesn't work in Swift.

Changes:
- [x] Move request strategy creation out of ZMSyncStrategy
- [x] Remove any direct dependency on a specific request strategy 
- [x] Re-write tests not to depend on OCMock

### CODE REVIEW

**Comment by typfel** [comment]
passed: *regression automation test*

File: Source/UserSession/ZMUserSession.swift [hunk 1]
```diff
@@ -513,3 +519,16 @@ extension ZMUserSession: URLActionProcessor {
         urlActionProcessors?.forEach({ $0.process(urlAction: urlAction, delegate: delegate)} )
     }
 }
+
+// TODO jacob temporary solution while refactoring
```
Comment (David-Henner) [inline Source/UserSession/ZMUserSession.swift:523]: is this intended to stay?

File: Tests/Source/Synchronization/MockRequestStrategyFactory.swift [hunk 2]
```diff
@@ -0,0 +1,24 @@
+//
+//  MockRequestStrategyFactory.swift
+//  WireSyncEngine-iOS-Tests
+//
+//  Created by Jacob Persson on 13.11.20.
+//  Copyright © 2020 Zeta Project Gmbh. All rights reserved.
+//
```
Comment (David-Henner) [inline Tests/Source/Synchronization/MockRequestStrategyFactory.swift:7]: is this the right header ?

File: Source/UserSession/ZMUserSession.swift (hunk 1 above)
Comment (typfel) [reply Source/UserSession/ZMUserSession.swift:523]: Thanks, I'll re-evaluate this.

File: Tests/Source/Synchronization/MockRequestStrategyFactory.swift (hunk 2 above)
Comment (typfel) [reply Tests/Source/Synchronization/MockRequestStrategyFactory.swift:7]: Nope, I'll update

**Review by KaterinaWire — APPROVED** [review]

File: Source/UserSession/ZMUserSession.swift (hunk 1 above)
Comment (typfel) [reply Source/UserSession/ZMUserSession.swift:523]: I would keep it for now. IMO to properly fix this I would need to break out the update event processing from the `ZMSyncStrategy` which is too large of a refactoring for this PR.

File: Tests/Source/Synchronization/MockRequestStrategyFactory.swift (hunk 2 above)
Comment (typfel) [reply Tests/Source/Synchronization/MockRequestStrategyFactory.swift]: Done

**Review by David-Henner — APPROVED** [review]

**Comment by typfel** [comment]
@zenkins release patch

**Pull request merged**